docs/source/getting_started/active_learning.rst

```diff
@@ -0,0 +1,164 @@
+
+Next, you will need to initialize the `ApiWorkflowClient` and the `ActiveLearningAgent`
+
+.. code-block:: Python
+
+    import lightly
+    from lightly.api import ApiWorkflowClient
+    from lightly.active_learning import ActiveLearningAgent
+
+    api_client = ApiWorkflowClient(dataset_id='xyz', token='123')
+    al_agent = ActiveLearningAgent(api_client) 
+
+
+.. note::
+
+   It may not always be a good idea to sample from the full dataset. For example,
+   it could be that a large portion of the images is blurry. In that case, it's 
+   possible to create a tag in the web-app which only contains the sharp images
+   and tell the `ActiveLearningAgent` to only sample from this tag. To do so, set
+   the `query_tag_name` argument in the constructor.
+
+Let's configure the sampling request and request an initial selection next:
+
+.. code-block:: Python
+
+   from lightly.active_learning import SamplerConfig
+   from lightly.openapi_generated.swagger_client import SamplingMethod
+
+   # we want an initial pool of 100 images
+   config = SamplerConfig(n_samples=100, method=SamplingMethod.CORESET, name='initial-selection')
+   initial_selection = al_agent.query(sampler_config)
+
+   assert len(initial_selection) == 100
+
+The query returns the list of filenames corresponding to the initial selection. Additionally, you
+will find that a tag has been created in the web-app under the name "initial-selection".
+Head there to scroll through the samples and download the selected images before annotating them.
+
+
+Active Learning Step
+~~~~~~~~~~~~~~~~~~~~~~
+
+After you have annotated your initial selection of images, you can train a model
+on them. The trained model can then be used to figure out, with which images it 
+has problems. These images can then be added to the labeled dataset.
+
+To do active learning with Lightly, you will need the `ApiWorkflowClient` and `ActiveLearningAgent` from before.
+If you have to re-initialize them, make sure to set the `pre_selected_tag_name` to your
+current selection (if this is the first iteration, this is the name you have passed 
+to the sampler config when doing the initial selection). Note, that if you don't 
+have to re-initialize them, the tracking of the tags is taken care of for you.
+
+.. code-block:: Python
+
+   # re-initializing the ApiWorkflowClient and ActiveLearningAgent
+    api_client = ApiWorkflowClient(dataset_id='xyz', token='123')
+    al_agent = ActiveLearningAgent(api_client, preselected_tag_name='initial-selection')
+
+The next part is what differentiates active learning from simple subsampling; the
+trained model is used to get predictions on the unlabeled data and the sampler then
+decides based on these predictions. To get a list of all filenames in the unlabeled set,
+you can simply call
+
+.. code-block:: Python
+
+   # get all filenames in the unlabeled set
+   unlabeled_set = al_agent.unlabeled_set
+
+Use this list to get predictions on the unlabeled images.
```

Comment:
I would say here, that the user has to use the own classifier to run the predictions. E.g. add

```python
your_classifier.train(al_agent.labeled_set)
predictions = your_classifer.predict_proba(al_agent.unlabeled_set)
```

Reply:
Hm I'm not sure this is so informative. I mean you cannot simply call `your_classifier.predict_proba` on a set of filenames, no? I agree, however, that this part might need clarification. Do you have any other ideas?

Reply:
```python
dataset_labeled = your_dataset.subset_from_filenames(agent.labeled_set)
your_classifier.train(dataset_labeled)
dataset_unlabeled = your_dataset.subset_from_filenames(agent.unlabeled)
predictions = your_classifer.predict_proba(dataset_unlabeled)
```

We could also add such a subset_from_filenames method to the LightlyDataset, we need it latest for the AL-tutorial with images.

Reply:
That's a good idea. I'd propose something like this:

```python
dataset = LightlyDataset.subset_from_filenames(input_dir, agent.unlabeled_set)
dataloader = torch.utils.data.DataLoader(dataset, shuffle=False, batch_size=1)
```

I will create an issue for this. Let's not add it to the docs before we have it implemented.